crates/rpc/engine/mod.rs

```diff
@@ -102,22 +103,25 @@ pub fn new_payload_v3(
         return Ok(PayloadStatus::syncing());
     }
 
-    // Fetch parent block header and validate current header
-    if let Some(parent_header) = storage.get_block_header(block.header.number.saturating_sub(1))? {
-        if !validate_block_header(&block.header, &parent_header) {
-            return Ok(PayloadStatus::invalid_with_hash(
-                parent_header.compute_block_hash(),
-            ));
-        }
-    } else {
-        return Ok(PayloadStatus::syncing());
-    }
+    let latest_valid_hash = latest_valid_hash(&storage).map_err(|_| RpcErr::Internal)?;
```

Comment:
```suggestion
    let latest_valid_hash = latest_valid_hash(&storage)?;
```
Mapping no longer needed